src/linter/SlangLinter.ts

```diff
@@ -0,0 +1,125 @@
+import * as vscode from 'vscode';
+import * as child from 'child_process';
+import * as path from 'path';
+import * as process from 'process';
+import BaseLinter from './BaseLinter';
+
+let isWindows = process.platform === 'win32';
+
+export default class SlangLinter extends BaseLinter {
+  private configuration: vscode.WorkspaceConfiguration;
+  private linterInstalledPath: string;
+  private arguments: string;
+  private includePath: string[];
+  private runAtFileLocation: boolean;
+  private useWSL: boolean;
+
+  constructor(diagnosticCollection: vscode.DiagnosticCollection, logger: vscode.LogOutputChannel) {
+    super('slang', diagnosticCollection, logger);
+    vscode.workspace.onDidChangeConfiguration(() => {
+      this.updateConfig();
+    });
+    this.updateConfig();
+  }
+
+  private updateConfig() {
+    this.linterInstalledPath = <string>(
+      vscode.workspace.getConfiguration().get('verilog.linting.path')
+    );
+    this.configuration = vscode.workspace.getConfiguration('verilog.linting.slang');
+    this.arguments = <string>this.configuration.get('arguments');
+    let path = <string[]>this.configuration.get('includePath');
+    this.includePath = path.map((includePath: string) => this.resolvePath(includePath));
+    this.runAtFileLocation = <boolean>this.configuration.get('runAtFileLocation');
+    this.useWSL = <boolean>this.configuration.get('useWSL');
+  }
+
+  protected convertToSeverity(severityString: string) {
+    if (severityString.startsWith('error')) {
+      return vscode.DiagnosticSeverity.Error;
+    } else if (severityString.startsWith('warning')) {
+      return vscode.DiagnosticSeverity.Warning;
+    }
+    return vscode.DiagnosticSeverity.Information;
+  }
+
+  private convertToWslPath(inputPath: string): string {
+    let cmd: string = `wsl wslpath '${inputPath}'`;
+    return child.execSync(cmd, {}).toString().replace(/\r?\n/g, '');
+  }
+
+  protected lint(doc: vscode.TextDocument) {
+    let docUri: string = doc.uri.fsPath;
+    let docFolder: string = path.dirname(docUri);
+    if (isWindows) {
+      if (this.useWSL) {
+        docUri = this.convertToWslPath(docUri);
+        this.logger.info(`Rewrote docUri to ${docUri} for WSL`);
+
+        docFolder = this.convertToWslPath(docFolder);
+        this.logger.info(`Rewrote docFolder to ${docFolder} for WSL`);
+      } else {
+        docUri = docUri.replace(/\\/g, '/');
+        docFolder = docFolder.replace(/\\/g, '/');
+      }
+    }
+
+    let slang: string = isWindows
+      ? this.useWSL
+        ? 'wsl slang'
+        : 'slang.exe'
+      : 'slang';
+
+    let binPath = path.join(this.linterInstalledPath, slang);
+    let args: string[] = [];
+    args.push('--single-unit');
+    args.push(`-I ${docFolder}`);
+    args = args.concat(this.includePath.map((path: string) => `-I ${path}`));
+    args.push(this.arguments);
+    args.push(`"${docUri}"`);
+    args.push('-Weverything', '--strict-driver-checking'); // Slang all warnings check options
```

Comment:
- Those arguments seem optional when linting with slang.
I think they should be in the `default` of the `verilog.linting.slang.arguments` section instead of the `args` because some users don't want to check all the warnings.
- We don't need `--lint-only` option?

Reply:
https://github.com/mshr-h/vscode-verilog-hdl-support/pull/387#issuecomment-1399183870

Sorry, I could not reply in comment.

How do you action to new option ?

Reply:
@retar-kamuy 
You don't need to put --lint-only option.

Reply:
Hi, @mshr-h san
Slang is a pure lint check tool, unlike Verilator which outputs code.

In Slang, the difference between when attached `--lint-only` and not attached the option is the check warnings rule.
By default (= not attached `--lint-only`), Slang is checked rules of integrity between instances in the elaborate phase, after the lint phase (Refer [slang command-line-ref](https://sv-lang.com/command-line-ref.html#Actions)).
Slang attached `--lint-only` is checked only syntax rules of single instance in the lint phase.

I think some users would like to elaborate phase checks by having multiple modules, read into slang by open in the editor and other codes included in that code(=readed by `+incdir` option).
I think those users would get stumped if they wrote that `--lint-only` option in `SlangLinter.ts`.

So I think don't need  `--lint-only` option.

If you have any other questions, feel free to ask. I want to help you even a little.

Reply:
Ok, I got it.
Can you remove the `args.push('--single-unit');` and `args.push('-Weverything', '--strict-driver-checking');`?

Reply:
Sure.
[The link](https://github.com/retar-kamuy/vscode-verilog-hdl-support/commit/0229265d76de329bc73ac5f5f3c307170f8b153f) that I made for pre-fix code after removing the lines.
If this code is sure, what should I do? I've never done the PR before, so I don't know how to fix a review. I'm sorry, but I would like you to tell me the next flow.

Reply:
You just need to commit the change to the PR branch.

Reply:
OK. I committed the chage.